server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1641,6 +1641,10 @@ public TranslogGeneration getMinGenerationForSeqNo(final long seqNo) {
      * @throws IOException if an I/O exception occurred during any file operations
      */
     public void rollGeneration() throws IOException {
+        //make sure we move most of the data to disk outside of the writeLock
```

Comment:
can you add a space after each `//` and what about rephrasing to:
```
make sure we move most of the data to disk outside of the writeLock in order to reduce the time the lock is held since it's blocking all threads
```

Reply:
Thank you for your advice! ^_^